### PR TITLE
Add configureBaker support

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Support `configureBaker` transactions

## Changes

Added serialization for `configureBaker`.
Added a function to generate baker keys.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/common/src/serializationHelpers.ts [hunk 1]
```diff
@@ -600,9 +600,63 @@ export function serializeConfigureDelegationPayload(
...
+const serializeVerifyKeys = serializeFromSpec<BakerKeysWithProofs>({
+    electionVerifyKey: encodeHexString,
+    proofElection: encodeHexString,
+    signatureVerifyKey: encodeHexString,
+    proofSig: encodeHexString,
+    aggregationVerifyKey: encodeHexString,
+    proofAggregation: encodeHexString,
+});
+
+const serializeUrl = (url: string) => {
```
Comment (rasmus-kirk) [inline packages/common/src/serializationHelpers.ts:623]: Same question as the related comment, should we have an `UrlString`?

**Review by rasmus-kirk — CHANGES_REQUESTED** [review]

File: packages/common/src/types.ts [hunk 2]
```diff
@@ -1185,6 +1185,28 @@ export interface UpdateCredentialsPayload {
...
+    stake?: CcdAmount;
+    /* should earnings from baking be added to staked amount  */
+    restakeEarnings?: boolean;
+    openForDelegation?: OpenStatus;
+    keys?: BakerKeysWithProofs;
+    metadataUrl?: string;
+    transactionFeeCommission?: number;
+    bakingRewardCommission?: number;
+    finalizationRewardCommission?: number;
+}
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/common/src/types.ts:1208]: They are settings, so only if you want to configure a setting, do you include it.

File: packages/common/src/serializationHelpers.ts [hunk 3]
```diff
@@ -600,9 +600,63 @@ export function serializeConfigureDelegationPayload(
...
+    const serializedPayload = serializeFromSpec(
+        configureDelegationSerializationSpec
+    )(payload);
+
+    return Buffer.concat([bitmap, serializedPayload]);
+}
+
+function encodeHexString(s: string): Buffer {
+    return Buffer.from(s, 'hex');
+}
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/common/src/serializationHelpers.ts:612]: I think it is a serializationHelper, but I'll export/document.

File: packages/rust-bindings/src/external_functions.rs [hunk 4]
```diff
@@ -330,3 +330,12 @@ pub fn serialize_credential_deployment_payload_ext(
     serialize_credential_deployment_payload_aux(signatures_vec, unsigned_info)
         .map_err(|e| format!("Unable to get credential deployment payload due to: {}", e))
 }
+
+#[wasm_bindgen(js_name = generateBakerKeys)]
+pub fn generate_baker_keys_ext(sender: &str) -> String {
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/rust-bindings/src/external_functions.rs:335]: Sure, I've done it for a couple other functions also.

**Review by rasmus-kirk — APPROVED** [review]

**Pull request merged**